### PR TITLE
SaiDeepak project summary alignment BMDashboard

### PR DESCRIPTION
**Pull request by SaiDeepak-NEU**
# Description
This PR is regarding the alignment of the project summary list which is off in the bmdashboard of Phase 2 and also making the project summary page responsive.

## Related PRS (if any):
This is only a frontend PR and can be tested using the development branch in the backend
…

## Main changes explained:
- Modified the ProjectSummary.jsx file
- Added some styling in BMDashboard.css
…

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin or owner user
5. navigate to /bmdashboard and login again
6. verify that the alignment of the content in the project summaries is in place and is not looking off like before. Please input any suggestions for any particular layout or design changes which would make the list and info look appealing.
7. Also, please verify the responsiveness of the table data by inspecting the page and verify alignment for different screen sizes.

## Screenshots or videos of changes:
Before Pic: 
Screenshot (103)
After alignment correction:
Screenshot (102)

## Note:
To test the responsiveness, right click and then click on inspect and then select mobile view and select different screen sizes and check alignment.

### CODE REVIEW

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I reviewed your PR and all the changes work as intended. I also tested the responsiveness with different screen sizes. Please refer the video attached.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/67577022/8b0be46d-d677-4cf9-91f8-e99dc80115ee

**Review by skaush21 — COMMENTED** [review]
* logged in as admin user
* navigated to /bmdashboard, logged in again
* verified alignment
* alignment of the table looks good on most devices except the galaxy fold, its more compacted and the text overlaps. 
* when switching from desktop/ipad to mobile the style/layout of the summary changes to a longer form. if that can be resized to appear like it does on the desktop that would be great. if not the current layout works just fine.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/141872402/a715834f-aade-4980-97ed-5b654f775c3e

**Review by chrischenlixing — DISMISSED** [review]
image
Hi this looks like it works as intended. Even when I try to shrink the window.

**Review by swathyjayaseelan — DISMISSED** [review]
The changes look good. The alignment of the page is responsive at all screen sizes.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/13806470/7fdf33bf-ec52-4240-b9b0-184254fbac2e

**Review by ShengjieMao — DISMISSED** [review]
The responsive page looks great in alignment. Toggling page doesn't affect the appearance. Great work.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/56333697/d656b24e-7fef-4b9c-98fc-3a707bbb18c1

**Review by FyisFe — CHANGES_REQUESTED** [review]
Tested with mobile and desktop view, the alignments looks great. However, there are still some bugs need to be fixed.

image

image

File: src/components/BMDashboard/Projects/ProjectSummary.jsx [hunk 1]
```diff
@@ -14,75 +14,75 @@ function ProjectSummary({ project }) {
   return (
     <div className="project-summary_content">
       <Row className="project-summary_header mx-auto">
-        <h2>{project.name} summary</h2>
+        <h2>{project.name} Summary</h2>
       </Row>
       <Row className="project-summary_item mx-auto">
-        <Col xs="5">
+        <Col xs={window.innerWidth < 700 ? '6' : '4'}>
```
[the remark this answers is not in the document]
Comment (FyisFe) [reply src/components/BMDashboard/Projects/ProjectSummary.jsx:20]: A bug here is directly accessing `window.innerWidth` in the JSX, which does not respond to window resizing after the initial render, use the `useState` and `useEffect` hooks from React to create a responsive layout that reacts to window size changes.

You can navigate to this tag, and when resizing window, it doesn't change from `col-6` to `col-4` and vice versa
image

**Review by XiaohanMeng — DISMISSED** [review]
Hi, I reviewed the PR and everything is functioning correctly. The responsiveness and alignment look great. Good job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113215317/71ffdb0b-5bad-417d-9c7c-c49d384cb053

File: src/components/BMDashboard/Projects/ProjectSummary.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (SaiDeepak-NEU) [reply src/components/BMDashboard/Projects/ProjectSummary.jsx:20]: Thank you for checking the code and suggesting appropriate changes. I appreciate your comment and I certainly agree that the change you suggested is a better approach. I will make the changes and will incorporate them as soon as possible.

**Review by FyisFe — DISMISSED** [review]
LGTM! Thanks for your contribution.

**Review by Gugor99 — APPROVED** [review]
### Report 
Hi SaiDeepak,
I checked out your implementation and it seems to be working as intended! Here are the steps I took: I logged in as admin, then navigated to /bmdashboard and logged in again. The layout was looking nicely both with phone and big screen size screens, well done! Please see screenshots for reference
### Screenshots
Cattura1
Cattura2
Cattura3

**Review by ClemarNunes — DISMISSED** [review]
I tested your PR and worked as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/134466562/3bd8e954-e00d-4a03-9119-a80739b48625

**Review by tdkent — CHANGES_REQUESTED** [review]
The page alignment and overall page layout looks good at standard desktop sizes. As other reviewers have pointed out, the layout is less readable for mobile views. This is mainly due to the left padding constricting the summary container. I played around with the styling for a few minutes and found that adding a rule such as `padding: 0` to `.projects-list` in **BMDashboard.css** eliminates most of the excess padding of the container itself. The container should also be centered.

There is padding inside the container pushing the text inwards that I think can be toned down/removed for mobile views.

Example of suggested improved mobile layout: 

Screenshot 2024-03-03 at 9 47 44 AM

This is just a quick example, and its causing some new issues with the text formatting, but I think it's closer to how the inner padding should be formatted. Note that I had to add a padding rule to each `<Col>` tag in **ProjectSummary.jsx** to make this work.

Screenshot 2024-03-03 at 9 52 17 AM

**Review by SanketKaware — DISMISSED** [review]
This change looks good to me.  /bmdashboard page alignment is correct.

I verified alignment for different screen sizes. 

Approving this change, **but my suggestion is** to check the alignment of "Select a project" and "Go to project dashboard" which are not aligned at the center.


Screenshot 2024-03-22 at 6 15 55 PM

Screenshot 2024-03-22 at 6 19 23 PM



**Please see my video review for reference**

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/17793708/9c67fa16-d5f7-4d3d-86c2-4cdb664068b0

**Review by tdkent — APPROVED** [review]
Update to padding improves mobile view.

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**